.github/workflows/ci.yml

```diff
@@ -21,9 +21,9 @@ jobs:
       fail-fast: false
       matrix:
         python:
-          - name: CPython 3.9
-            version: 3.9
-            action: 3.9
+          - name: CPython 3.7
+            version: 3.7
+            action: 3.7
```

Comment:
Why downgrade from 3.9 to 3.7?

Reply:
i get this error with 3.9

poetry.utils.env.NoCompatiblePythonVersionFound: Poetry was unable to find a compatible version. If you have one, you can explicitly use it via the "env use" command. 30

ill try to fix it.

Reply:
copy paste from chat:

so i had downgraded python 3.9 to 3.7.
I am working on getting it back and running for 3.9 but im hitting a block.
so i changed it back to 3.9 in the ci, and in poetry i changed python to ^3.9.
CI runs through and makes the artifact, but the testing of the artifact fails.  I eventually pulled the artifact to test it myself and got this error.
did some research to see that time.clock is deprecated in python 3.8.  Im not sure how this was working before.  I read that the fix is to change time.clock -> time.time.
The issue is that the time.clock is in the .venv, (.venv\Lib\site-packages\sunspec\core\modbus\client.py) so im not exactly sure what to do about that.

Reply:
Appears that updating pysunspec to v2.1.1 has alleviated the issue. Still need to test.